src/Pos/Wallet/Web/Tracking.hs

```diff
@@ -273,7 +273,6 @@ applyModifierToWSet
     -> m ()
 applyModifierToWSet wAddr newTip mapModifier = do
     -- TODO maybe do it as one acid-state transaction.
-    mapM_ WS.removeWAddress (MM.deletions mapModifier)
```

Comment:
Why not? Addresses deletions occur only when address is spent (from blockchain's point of view). That's exactly what we don't want to do since now.